### DISCUSSION TITLE
request to consider integrating routing lib

### DISCUSSION THREAD

**Discussion by devhandler** [alpinejs/alpine#3581] [3. Feature ideas]
alpine is so intuitive to learn and super useful. thank you! do you mind considering integrating routing features like this one:
https://github.com/shaunlee/alpinejs-router

**Comment by SimoTod** [comment]
Why do you need it in the core lib? It's been written and maintained by a different developer. 

If brought into the code, the Alpine author will take ownership and, more important, will have to maintain the feature and fix bugs.

**Reply by devhandler** [reply]
an alternative solu could be providing a recommendation in the readme which routing lib to use. that would greatly help new adopters identify a start point for routing.

**Reply by ekwoka** [reply]
Right now, routing isn't in the goals of Alpine.

It is not an SPA framework.

It's for easily extending server rendered multi page apps with reactivity.

A router is fundamentally against that goal.

It's a fun little toy, I even made my own Alpine router, but it's very much the wrong tool.

**Reply by bora89** [reply]
Well if the goal only to provide reactivity to some server side rendering - then Alpine has very limited utility.
I noticed that adoption rate has slowed down which you can see from YouTube videos - all of them older than a year. 
Although it is relatively simple to learn it, but knowing those limitations should be considered before investing time on both sides.

**Reply by ekwoka** [reply]
> Well if the goal only to provide reactivity to some server side rendering - then Alpine has very limited utility.

Not so sure I see that.

There are tons of SSR systems out there, in every language under the sun.

This just isn't a full-stack UI framework.

It's not what it's for.

Alpine isn't competing with React, or Vue, or Svelte, or Solid, or whatever.

**Reply by bora89** [reply]
Hi @ekwoka , I think I did not understand that sentence quite fully
> It's a fun little toy, I even made my own Alpine router, but it's very much the wrong tool.

As if it is wrong to use Alpine with a router? Please explain. Btw I understand that including it to lib is not a choice at all. All I think we wanted is to have a broader explanation what it is for and more real examples in the documentation

**Reply by ekwoka** [reply]
Alpine isn't a full SPA UI Library. That's why it's the wrong tool.

I kind of find that the docus are pretty clear what the purpose is.

You provide functional HTML to the client. Alpine looks at it and hooks up reactivity.

That's the purpose.

So it's fantastic for people using PHP, Liquid, Python, Go, Rust, .NET, etc to server render content.

**Comment by boustanihani** [comment]


**Reply by meirlamdan** [reply]
I recommend you check out vue.js...

**Comment by boustanihani** [comment]
Here is a good router for Alpine => https://github.com/pinecone-router/router
